repository-azure/src/main/java/io/aiven/elasticsearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.elasticsearch.repositories.azure;
+
+import java.io.InputStream;
+
+import io.aiven.elasticsearch.repositories.CommonSettings;
+
+import org.elasticsearch.common.settings.SecureSetting;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+
+import static io.aiven.elasticsearch.repositories.CommonSettings.KeystoreSettings.checkSettings;
+import static io.aiven.elasticsearch.repositories.CommonSettings.KeystoreSettings.withPrefix;
+
+public class AzureStorageSettings implements CommonSettings.KeystoreSettings {
+
+    static final String AZURE_CONNECTION_STRING_TEMPLATE =
+            "DefaultEndpointsProtocol=https;AccountName=%s;AccountKey=%s";
+
+    static final Setting<Integer> AZURE_HTTP_POOL_MIN_THREADS =
+            Setting.intSetting(withPrefix("azure.http.thread_pool.min"), 1, Setting.Property.NodeScope);
+
+    static final Setting<Integer> AZURE_HTTP_POOL_MAX_THREADS =
+            Setting.intSetting(withPrefix("azure.http.thread_pool.max"),
+                    Runtime.getRuntime().availableProcessors() / 2, Setting.Property.NodeScope);
+
+    static final Setting<TimeValue> AZURE_HTTP_POOL_KEEP_ALIVE =
+            Setting.timeSetting(withPrefix("azure.http.thread_pool.keep_alive"), TimeValue.timeValueSeconds(30L),
+                    Setting.Property.NodeScope);
+
+
+    public static final Setting<InputStream> PUBLIC_KEY_FILE =
+            SecureSetting.secureFile(withPrefix("azure.public_key_file"), null);
+
+    public static final Setting<InputStream> PRIVATE_KEY_FILE =
+            SecureSetting.secureFile(withPrefix("azure.private_key_file"), null);
+
+    public static final Setting<SecureString> AZURE_ACCOUNT =
```

Comment:
I didn't find an answer in code, so maybe you help me. What happened if there is an azure account in settings, but it is not valid. How did we find out that it is not valid?

Reply:
When ES creates repo it validates it (build in functionality).